tests/test_urban_nature_access.py

```diff
@@ -194,129 +268,199 @@ def test_greenspace_budgets(self):
+            numpy.testing.assert_allclose(
+                aligned_lulc_raster_info['pixel_size'],
+                aligned_population_raster_info['pixel_size'])
+            numpy.testing.assert_allclose(
+                aligned_lulc_raster_info['raster_size'],
+                aligned_population_raster_info['raster_size'])
+            numpy.testing.assert_allclose(
+                aligned_lulc_raster_info['geotransform'],
+                aligned_population_raster_info['geotransform'])
+            numpy.testing.assert_allclose(
+                aligned_lulc_raster_info['bounding_box'],
+                aligned_population_raster_info['bounding_box'])
+
+            # Check that we're getting the appropriate summary values in the
+            # admin units vector.
+            admin_vector_path = os.path.join(
+                args['workspace_dir'], 'output',
+                f"admin_units_{args['results_suffix']}.gpkg")
+            admin_vector = gdal.OpenEx(admin_vector_path)
+            admin_layer = admin_vector.GetLayer()
+            self.assertEqual(admin_layer.GetFeatureCount(), 1)
+
+            # expected field values from eyeballing the results; random seed = 1
+            expected_values = {
+                'SUP_DEMadm_cap': -17.9078,
+                'Pund_adm': 4094.012207,
+                'Povr_adm': 981.987549,
+            }
+            admin_feature = admin_layer.GetFeature(1)
+            self.assertEqual(
+                expected_values.keys(),
+                admin_feature.items().keys()
+            )
+            for fieldname, expected_value in expected_values.items():
+                numpy.testing.assert_allclose(
+                    admin_feature.GetField(fieldname), expected_value)
+
+            # The sum of the under-and-oversupplied populations should be equal
+            # to the total population count.
+            population_array = pygeoprocessing.raster_to_numpy_array(
+                args['population_raster_path'])
+            numpy.testing.assert_allclose(
+                (expected_values['Pund_adm'] + expected_values['Povr_adm']),
+                population_array.sum())
+
+            admin_vector = None
+            admin_layer = None
+
+            shutil.rmtree(args['workspace_dir'])
+
+    def test_split_greenspace_unique_distances(self):
+        from natcap.invest import urban_nature_access
+
+        args = _build_model_args(self.workspace_dir)
+
+        # The split greenspace feature requires an extra column in the
+        # attribute table.
+        attribute_table = pandas.read_csv(args['lulc_attribute_table'])
+        new_search_radius_values = {
+            value: 30*value for value in range(1, 10, 2)}
+        attribute_table['search_radius_m'] = attribute_table['lucode'].map(
+            new_search_radius_values)
+        attribute_table.to_csv(args['lulc_attribute_table'], index=False)
+        # TODO: check with science team about enabling split greenspace.
```

Comment:
That's interesting. I'm not plugged in to the details here, but I wonder if there needs to be an explicit enable/disable switch? With the way I think you have it set up, if the table includes the search radius column with different radii, then that is indicating a `split greenspace`. That seems right to me, with a LOGGER INFO indication of which mode is being used. But yea, it'd have to be articulated well in the Uses Guide.

Reply:
Yeah, there might not need to be any more, if we assume that the presence of multiple search radii triggers the split greenspace mode!

I'll need to work on this a little more and I'll get back to you soon.

Reply:
OK! I've implemented this in https://github.com/phargogh/invest/commit/52fbbd565b776ba91b79eab0bf27de957f1d2803.  The behavior now is:

* The `search_radius_m` column is always required in the attributes table
* If the search radii are all the same, we treat that as "unified greenspace" and all greenspace lucodes will be processed together.  An INFO message is logged describing what's up.
* If there are multiple detected search radii, we use "split greenspace" mode and each individual LULC code is processed seprately (per request of the science team).  an INFO message is logged describing what's up.

As a part of this I also factored the kernel creation and population convolution out of the 2SFCA loop since there isn't necessarily a 1:1 mapping of search radii to greenspace lucodes.